### PR TITLE
Port ProximalAdagrad

### PR DESCRIPTION
**Pull request by WindQAQ**
Closes https://github.com/tensorflow/addons/issues/591.

### CODE REVIEW

File: tensorflow_addons/optimizers/proximal_adagrad.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        if initial_accumulator_value < 0.0:
+            raise ValueError("`initial_accumulator_value` must be non-negative.")
+        if l1_regularization_strength < 0.0:
+            raise ValueError("`l1_regularization_strength` must be non-negative.")
+        if l2_regularization_strength < 0.0:
+            raise ValueError("`l2_regularization_strength` must be non-negative.")
+        super().__init__(name, **kwargs)
+        self._set_hyper("learning_rate", kwargs.get("lr", learning_rate))
+        self.l1_regularization_strength = l1_regularization_strength
+        self.l2_regularization_strength = l2_regularization_strength
```
Comment (facaiy) [inline tensorflow_addons/optimizers/proximal_adagrad.py:80]: How about creating l1_regularxx and l2_regularxx as hyper-parameter?

File: tensorflow_addons/optimizers/proximal_adagrad.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Proximal Adagrad optimizer."""
+
+import tensorflow as tf
```
Comment (facaiy) [inline tensorflow_addons/optimizers/proximal_adagrad.py:17]:
```suggestion
from typing import Callable, Union

import tensorflow as tf
from typeguard import typechecked

from tensorflow_addons.utils.types import FloatTensorLike
```

File: tensorflow_addons/optimizers/proximal_adagrad.py (hunk 1 above)
Comment (WindQAQ) [reply tensorflow_addons/optimizers/proximal_adagrad.py:80]: Will do. Just copy paste the name from `compat.v1`. Maybe `l1` and `l2` are more suitable?

https://www.tensorflow.org/api_docs/python/tf/keras/regularizers/L1L2

Comment (facaiy) [reply tensorflow_addons/optimizers/proximal_adagrad.py]: I agree, `l1` and `l2` are more concise :-)

Comment (facaiy) [reply tensorflow_addons/optimizers/proximal_adagrad.py]: Tzu-Wei, perhaps we could use `self._set_hyper` for `l1` and `l2`, what do you think?

Comment (WindQAQ) [reply tensorflow_addons/optimizers/proximal_adagrad.py]: of course. let me update it real quick :-)

Comment (WindQAQ) [reply tensorflow_addons/optimizers/proximal_adagrad.py]: Seems that `_set_hyper`/`_get_hyper` supports more types of inputs.

**Comment by WindQAQ** [comment]
Umm, sorry for the confusion, but `tf.keras.optimizers.Ftrl` use `l{1,2}_regularization_strength` instead :-(
I'm going to change it back.

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizer_v2/ftrl.py#L47-L50

**Pull request merged**